src/Makefile-env.am

```diff
@@ -87,6 +86,18 @@ AM_COMMON_CPPFLAGS = \
 	-DCEPH_PKGLIBDIR=\"${pkglibdir}\" \
 	-DGTEST_USE_OWN_TR1_TUPLE=0
 
+if LINUX
+AM_COMMON_CPPFLAGS += \
+	-D_REENTRANT
+endif
+
+if SOLARIS
+AM_COMMON_CPPFLAGS += \
+       -D_PTHREADS \
+       -D_POSIX_C_SOURCE
+	
+endif
```

Comment:
this may break compilation on OSX and my pending FreeBSD port. please avoid doing this